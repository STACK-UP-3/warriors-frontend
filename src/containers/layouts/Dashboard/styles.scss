/**
 * Google Fonts
 */
@import url('https://fonts.googleapis.com/css2?family=IBM+Plex+Sans+Condensed:wght@300;400;500;700&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Archivo:wght@400;500;700&display=swap');

html {
	overflow-y: scroll;
}

// Remove margin just before footer
main .row:last-child {
	margin-bottom: 0;
}

.bg-primary-color {
	background-color: #083b66;
}
