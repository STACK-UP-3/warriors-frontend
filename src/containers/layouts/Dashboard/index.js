import React from 'react';
import PropTypes from 'prop-types';

// Load styles
import './styles.scss';

// Load page components
import Header from '../../blocks/Header';
import Sidebar from '../../blocks/Sidebar';
import Footer from '../../blocks/Footer';

export default class Dashboard extends React.Component {
  constructor() {
    super();

    // // user must be logged in to access the dashboard
    // const token = localStorage.getItem('token');
    // if (!token) window.location.replace('/login');
  }

  render() {
    return (
      <div>
        <Header />
        <main>
          <div className="row">
            <div className="col s12 m3">
              <Sidebar />
            </div>
            <div className="col s12 m9">{this.props.children}</div>
          </div>
        </main>
        <Footer />
      </div>
    );
  }
}

/**
 * Define prop types for the Component
 */
Dashboard.propTypes = {
  children: PropTypes.node.isRequired,
};
