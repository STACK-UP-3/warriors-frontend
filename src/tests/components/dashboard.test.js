import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../../components/dashboard';

describe('*************** Testing the Dashboard  component ***************',()=>{
    it('Should render the dashboard page correctly', ()=>{
        const wrapper = shallow( <Dashboard  /> );
        expect(wrapper).toMatchSnapshot();
    });
});