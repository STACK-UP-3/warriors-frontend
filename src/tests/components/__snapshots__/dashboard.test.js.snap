// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the Dashboard  component *************** Should render the dashboard page correctly 1`] = `
<div>
  <NavBar />
  <main
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <div
      className="sideMenu"
    >
      <div
        className="fixed-sidebar container center hide-on-med-and-down"
      >
        <_default />
      </div>
      <div
        className="show-on-med-and-down container center"
      >
        <mobileSidebar />
      </div>
    </div>
    <div
      className="mainContent"
    >
      <h1>
        DashBoard
      </h1>
    </div>
  </main>
  <SideInitializer />
  <_default />
</div>
`;
